pkg/controller/configmap/configmap_controller.go

```diff
@@ -588,7 +588,7 @@ func (r *ReconcileConfigMap) isSecretChanged(opNsxSecret *corev1.Secret, opLbSec
 		}
 		ncpNsxSecret = nil
 	}
-	nsxSecretChanged := secretEqual(opNsxSecret, ncpNsxSecret)
+	nsxSecretChanged := secretNotEqual(opNsxSecret, ncpNsxSecret)
```

Comment:
I believe func name :secretEqual is not semantically correct according to function purpose.
I suggest it be changed to secretNotEqual although its intrinsic logic is correct.

Reply:
How about `isSecretChanged`?

Reply:
We already have a func named by:isSecretChanged
func (r *ReconcileConfigMap) isSecretChanged(opNsxSecret *corev1.Secret, opLbSecret *corev1.Secret) (bool, error) {
...

secretNotEqual func is called in this func. Moreover, this func is used to compare secret content. 

Otherwise, we change logic of secretEqual to make it return true when equal, it make more intuitive. 
In such way,

nsxSecretEqual := secretEqual(opNsxSecret, ncpNsxSecret)
if !nsxSecretEqual {
 		log.Info("nsx-secret is changed")
 		return true, nil
     .........